### PR TITLE
KAFKA-6988: Improve Kafka classpath length by using all files in dir instead of adding one by one

### PR DESCRIPTION
**Pull request by lkgendev**
Fixing Windows too long command due to classpath in case kafka is installed in a directory that has large length, also shorten the Unix classpath as it can have similar problem on docker

the contribution is my original work and I license the work to the project under the project's open source license.

### CODE REVIEW

**Comment by lkgendev** [comment]
retest this please

**Comment by lkgendev** [comment]
retest this please

File: bin/kafka-run-class.sh [hunk 1]
```diff
@@ -143,12 +143,9 @@ do
 done
 
 # classpath addition for release
-for file in "$base_dir"/libs/*;
-do
-  if should_include_file "$file"; then
-    CLASSPATH="$CLASSPATH":"$file"
-  fi
-done
+set -o noglob
```
Comment (ryannedolan) [inline bin/kafka-run-class.sh]: The noglob stuff is unnecessary here.

**Comment by harshach** [comment]
@lkgendev overall looks good to me. +1 after addressing pending comment.

**Comment by lkgendev** [comment]
@ryannedolan you are right, the quotes around the asterisk eliminates the need for noglob, removed this part in the second commit

**Comment by lkgendev** [comment]
retest this please

**Comment by OneCricketeer** [comment]
Doesn't just adding on a glob cause other packages to be loaded, if they exist? 

https://github.com/apache/kafka/blob/trunk/bin/kafka-run-class.sh#L34-L42

As a system administrator, I think the solution could be relocating the Kafka installation folder... You wouldn't lose any data in this process with a quick broker restart and keeping `logs.dirs` the same.

The larger problem would be the having the files in the logs folder being too long.

**Comment by lkgendev** [comment]
@cricket007 @ryannedolan the glob part have been removed in the second commit
about installation folder, that makes you OS dependent and limits the number of jars, once there are too many jars or jars with long names it will not work regardless of installation folder path, it also prevents using Kafka as a third party in other products that have a product directory, the large length of jars is not just in the additional jars but also in the regular jars

**Comment by OneCricketeer** [comment]
When I say glob, I mean putting `lib/*`... In other words, if `should_include_file` isn't necessary, should it be removed? Shouldn't these changes also be applied to Kafka Connect and the other sections of the script that loop over JAR files?

**Comment by lkgendev** [comment]
@cricket007 there is a difference between the lib/* and the other jar loops
In the lib/* all jar files in the directory are in use and so putting * is equivalent and loads the same set of jar files, however in the other loops, the loop take only specific pattern files such as "kafka-clients*.jar" etc, I am not sure if java has flags to take all files matching a pattern, so using * instead of such loops will take more files than currently used, but it is possible to change those loops and take all files in the directories
about `should_include_file`, if you can set a smaller set of jars in some cases I think it is better to do so and not load redundant files

**Comment by OneCricketeer** [comment]
> In the lib/* all jar files in the directory are in use

What I'm trying to say here is by not doing `should_include_file`, things that *should not be part of the CLASSPATH*, now will be. 

That's all.

And the other loops seem the same to me. If you tried to run Kafka Connect on Windows with a long directory path, you'd run into the same problems, so again, I think the solution would be to simply relocate your Kafka installation folder to like `C:\kafka`

**Comment by lkgendev** [comment]
@cricket007 You are right, following the suggested commit, the `should_include_file` will have no effect on the regular libs directory jars and all jar files in this directory will be included even if not with regex
However without this change in some platforms such as centos docker and windows there is a problem running the program depending on the installation path
That is up to the reviewers to decide if this regular expression is necessary, it is possible maybe to have another configuration variable that allows for choosing how to set the classpath

File: bin/kafka-run-class.sh (hunk 1 above)
Comment (lkgendev) [reply bin/kafka-run-class.sh]: done in second commit

**Comment by iamsivav** [comment]
I got struck by this issue when i try to set up my development environment in my windows laptop. 
It is necessary to incorporate this solution ASAP.  
The solution provided in this https://github.com/kafka-dev/kafka/issues/61#issue-323657794 issue worked for me!!!

**Pull request closed without merging**